pandas/io/formats/style.py

```diff
@@ -2551,16 +2551,16 @@ def hide_columns(
         Styler.hide: Hide the entire index / columns, or specific rows / columns.
         """
         warnings.warn(
-            "this method is deprecated in favour of `Styler.hide(axis='columns')`",
+            'this method is deprecated in favour of `Styler.hide(axis="columns")`',
             FutureWarning,
             stacklevel=find_stack_level(),
         )
-        return self.hide(axis=1, level=level, subset=subset, names=names)
+        return self.hide(axis="columns", level=level, subset=subset, names=names)
 
     def hide(
         self,
         subset: Subset | None = None,
-        axis: Axis = 0,
+        axis: Axis = "index",
```

Comment:
```suggestion
        axis: Axis = 0,
```
most methods with str and int input default to 0. leave as is for consistency.

Reply:
Most of my colleagues I know use pandas agree that using Axis as string greatly improves readability specially to those that work with a lot of code from other libraries. This notation was introduces later and this is why is more frequent on the code. I would counter-propose, if you agree, just to change the only other method in this module that initializes Axis = 0, which is `set_sticky()`.

Reply:
if `set_sticky` is not default `Axis=0` then it should be, we can make that a DOC change.

Of course your own code can include `axis="columns"` etc., that's why it's available. But changing the pandas internal code will have no impact on external code readability.

Reply:
You are right, it doesn't improve external code, but it does improve internal code readability. This is what this PR is all about, making pandas code better for users who want to dig in debugging and also to make it easier to contribute and make it even more awesome.

Reply:
pandas defaults Axis to integers in general.